### PR TITLE
master

### PR DESCRIPTION
**Pull request by mkotajny**
Task5

### CODE REVIEW

File: Task5/App.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import React, { useEffect, useState } from 'react';
+
+function App() {
```
Comment (10softpl) [inline Task5/App.js:3]: please move the fuel logic out of the main component of the Application (out of “App”) . I mean, please move the fuel-logic to some external component (like “FuelAlert”)

File: Task5/App.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  
+    var [fuel, setFeul] = useState(0);
```
Comment (10softpl) [inline Task5/App.js:5]: typo in the name of “setFeul”

File: Task5/App.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    var [alertText, setAlertTxt] = useState('Processing...');
```
Comment (10softpl) [inline Task5/App.js:6]: - I recommend name “setAlertText” (not “setAlertTxt”) to keep consistency with the name of alertText
-  consider to track of state of processing in a separate state value named ‘processing’ or ‘inProgress’ and for “in progress” state - render some nice spinner (instead of just showing message “Processing …”)

File: Task5/App.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React, { useEffect, useState } from 'react';
```
Comment (10softpl) [inline Task5/App.js:1]: please move the fuel logic out of the main component of the Application (out of “App”) . I other words - please move the fuel-logic to some external component (like “FuelAlert”)

File: Task5/App.js (hunk 2 above)
Comment (10softpl) [inline Task5/App.js:5]: typo in the name of “setFeul”
inappropriate default value for the “fuel” state, I recommend value of undefined

File: Task5/App.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+function App() {
+  
+    var [fuel, setFeul] = useState(0);
+    var [alertText, setAlertTxt] = useState('Processing...');
+
+    useEffect(() => {
+      fetch('https://new.world.com/fleet/121')
```
Comment (10softpl) [inline Task5/App.js:9]: - I recommend to replace hard coded 121 in the request with the property named “fleetId” (and invoke the Fuel component with fleetId=”121”)
- also make sure with the backend that this endpoint returns **monthly** value (not all history value), as the User Story defines alert for monthly fuel usage

File: Task5/App.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        .then(response => response.json())
+        .then(json => {
+          setFeul(json);
+          console.log(json);
```
Comment (10softpl) [inline Task5/App.js:13]: please remove the console.log

File: Task5/App.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        })
```
Comment (10softpl) [inline Task5/App.js:14]: please add any error handling for this request (.catch(...)), because without error handling - in case of any error - alertText state will stay “Processing…” forever

File: Task5/App.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    });
```
Comment (10softpl) [inline Task5/App.js:15]: please add empty array as a second parameter to this useEffect, because (during my manual tests) I see that this request has been sent too many times (on each rerender of the component), but should be run once (on 1st render only)

File: Task5/App.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+    useEffect(() => {
+      if (!fuel) setAlertTxt('Processing...'); else
```
Comment (10softpl) [inline Task5/App.js:18]: this line (imo) is not necessary as alertText already has a (default) value ‘Processing’, anyway check this please

File: Task5/App.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+      if (fuel.litres > 0) setAlertTxt('Need to buy more fuel'); else
```
Comment (10softpl) [inline Task5/App.js:19]: - in my opinion - the condition in the if statement is wrong. For example - this logic will alarm about lack of fuel in case of 5 litres of fuel-usage  (should alarm for litres > 10 base on User Story description). Please fix the logic.
- also please reproduce (manually test) of **many possible testing-scenarios** (including all possible edge cases) to eliminate bugs like this one

File: Task5/App.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+        .then(json => {
+          setFeul(json);
+          console.log(json);
+        })
+    });
+
+    useEffect(() => {
+      if (!fuel) setAlertTxt('Processing...'); else
+      if (fuel.litres > 0) setAlertTxt('Need to buy more fuel'); else 
+      setAlertTxt('All is fine');
```
Comment (10softpl) [inline Task5/App.js:20]: wrong condition (else → all fine). 
This logic will inform “all is fine” also if there will be no “litres” field (at all) in the API response. 
The safer approach might be - to inform “All fine” based on **direct condition** -  if (fuel.litres < 10)  … even better replace hard coded 10 with maxFuelUsage component property, and invoke this component with prop value maxFuelUsage=10

File: Task5/App.js [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+    }, [fuel])
```
Comment (10softpl) [inline Task5/App.js:21]: First useEffect is ended with semicolon, but this (the 2nd) useEffect is **not** ended with semicolon (format inconsistency). 
I recommend to install **Prettier** and configure it (set formatting rules)  on this project, so I will not have to remember about keeping formatting rules manually.

File: Task5/App.js [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+    return (
+      <div>
```
Comment (10softpl) [inline Task5/App.js:24]: you can remove div tag, as contains only one child - h1 tag

File: Task5/App.js [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+        <h1 style={alertText == 'Need to buy more fuel'? {color:"red"}:{}}>{alertText}</h1>
```
Comment (10softpl) [inline Task5/App.js:25]: - please avoid inline styling , and export styles into separate file (like fuel.styles.css)
- Regarding style of the alert (big red letters) - please work little bit more on the styling (search for inspiration on the Internet for nice stylings of alerts, error-snackbars etc) … even better - export (better styled) alert into separate component (named "Alert" or other similar name) for future reuse.
